java-checks/src/main/java/org/sonar/java/checks/FieldModifierCheck.java

```diff
@@ -43,18 +44,28 @@
   @Override
   public void visitNode(Tree tree) {
     ClassTree classTree = (ClassTree) tree;
-    for (Tree member : classTree.members()) {
-      if (member.is(Tree.Kind.VARIABLE) && hasNoVisibilityModifier((VariableTree) member)) {
+    classTree.members().stream()
+      .filter(FieldModifierCheck::isConsentWithCheck)
+      .forEach(member -> {
         IdentifierTree simpleName = ((VariableTree) member).simpleName();
         reportIssue(simpleName, "Explicitly declare the visibility for \"" + simpleName.name() + "\".");
-      }
-    }
+      });
   }
 
-  private static boolean hasNoVisibilityModifier(VariableTree member) {
-    ModifiersTree modifiers = member.modifiers();
+  private static boolean isConsentWithCheck(Tree member) {
```

Comment:
FP. will be handled by resolution of [SONARJAVA-958](https://jira.sonarsource.com/browse/SONARJAVA-958)